utils/prompt.go

```diff
@@ -0,0 +1,46 @@
+package utils
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+)
+
+// confirm prompts a user for a confirmation with message
+// and returns true with no error if input is "yes" or "y" (case-insensitive),
+// otherwise false.
+func Confirm(message string) (bool, error) {
+	fmt.Print(message)
+	yes := []string{"yes", "y"}
+	var input string
+	_, err := fmt.Scanln(&input)
+	if err != nil {
+		return false, err
+	}
+	input = strings.ToLower(input)
+	for _, valid := range yes {
+		if input == valid {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+func ConfirmFileOverwrite(filename string, ext string, assumeYes bool) (bool, error) {
+	if assumeYes {
+		return true, nil
+	}
+
+	filename = AddExtToFilename(filename, ext)
```

Comment:
q: just wondering why we don't just take `filename` and remove the `ext` var here, is there some context I'm missing where the extension is important?

Reply:
The reason is 'I wrote bad code' and hence this is the way it is. It will go away in a future refactor.
This has weird implications otherwise.